spark_log_parser/parsing_models/application_model_v2.py

```diff
@@ -111,76 +112,125 @@ def save_to_s3(self, saveDat, filepath, compress):
             dat = gzip.compress(json.dumps(saveDat).encode("utf-8"))
             s3.put_object(Bucket=bucket, Body=dat, Key=key + ".gz")
 
-        logging.info("Saved object to cloud: %s" % (key))
+        logger.info("Saved object to cloud: %s" % (key))
 
 
-DataType = TypeVar("DataType")
+SparkApplicationRawDataType = TypeVar("DataType")
+SparkApplicationLoaderKey = TypeVar("SparkApplicationLoaderKey")
 
 
-class AbstractSparkApplicationDataLoader(abc.ABC, Generic[DataType], DataLoader):
+class AbstractSparkApplicationDataLoader(abc.ABC,
+                                         Generic[SparkApplicationLoaderKey, SparkApplicationRawDataType],
+                                         DataLoader):
```

Comment:
By "CLI-oriented class", do you mean "a loader that only cares about `str`s as input"? If so, then yeah, we have that here, I think - https://github.com/synccomputingcode/spark_log_parser/pull/33/files#diff-0cf9281aad843e6aecfd873716309fbc86ad65573b24bfec4813588b817afc95R1093

We could take a delegation approach instead of an inheritance approach, certainly. My thinking was that this base class would provide the generic ability to load a `SparkApplication` against some input, and the concrete classes would describe what the expected input type is (and provide the specifics of "how" to load that type of input).